projects/RabbitMQ.Client/client/api/ConnectionFactoryBase.cs

```diff
@@ -49,12 +52,8 @@ public class ConnectionFactoryBase
         /// <returns>New instance of a <see cref="TcpClient"/>.</returns>
         public static ITcpClient DefaultSocketFactory(AddressFamily addressFamily)
         {
-            var socket = new Socket(addressFamily, SocketType.Stream, ProtocolType.Tcp)
-            {
-                NoDelay = true,
-                ReceiveBufferSize = 65536,
```

Comment:
You played around with the buffer size, what's the final verdict now?

Reply:
Leave it at default values, Pipelines have their own internal buffer and manage it themselves. Too big of a buffer at the socket level can confuse Pipelines and result in operations hanging.